seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java

```diff
@@ -173,7 +173,7 @@ private boolean likeExpr(LikeExpression likeExpression, Object[] inputFields) {
         Pattern pattern = Pattern.compile(regex);
         Matcher matcher = pattern.matcher(leftVal.toString());
 
-        return matcher.matches();
+        return matcher.matches() ^ likeExpression.isNot();
```

Comment:
Can we split `like` with `not like`? this function is `likeExpr` but is is used in `not like`, this is  strange .

Reply:
I don't think it's necessary to add a not like expression, because that's where the SQL engine expresses the match and mismatch logic. I'm referring to the IN expression logic